uqcsbot/advent.py

```diff
@@ -0,0 +1,365 @@
+import discord
+from discord.errors import InvalidArgument
+from discord.ext import commands
+from uqcsbot.bot import UQCSBot
+from uqcsbot.utils.command_utils import loading_status
+
+from argparse import ArgumentParser, Namespace
+from datetime import datetime, timedelta, timezone
+from requests.exceptions import RequestException
+from typing import Any, Callable, Dict, List, Optional
+from enum import Enum
+import os
+import io
+import requests
+import logging
+
+# Leaderboard API URL with placeholders for year and code.
+LEADERBOARD_URL = "https://adventofcode.com/{year}/leaderboard/private/view/{code}.json"
+# Session cookie (will expire in approx 30 days).
+SESSION_ID = os.environ.get("AOC_SESSION_ID")
```

Comment:
It might be worth adding a brief comment about how to get this running because there are a few steps (1. get a token by logging in and then inspecting a request, 2. make sure youve joined the leader board), unless this is documented elsewhere

Reply:
Yeah, I'll update this before merging